src/lib.ts

```diff
@@ -8,15 +9,50 @@ export interface EnvSchema {
         default: string;
     }
 }
-export function scanVars(dir: string) {
+export function scanVars(dir: string, useGitTracking: boolean = true) {
     const envVars = new Set<string>();
+    
+    // Common directories and files to ignore
+    const ignorePatterns = [
+        'node_modules',
+        '.git',
+        'dist',
+        'build',
+        'coverage',
+        '.next',
+        '.cache',
+        '.DS_Store'
+    ];
+
+    // Check if we can use Git for tracking files
+    const gitAvailable = useGitTracking && isGitAvailable(dir);
+    const gitTrackedFiles = gitAvailable ? getGitTrackedFiles(dir) : new Set<string>();
+
+    function shouldIgnore(pathName: string): boolean {
+        const baseName = path.basename(pathName);
+        return ignorePatterns.includes(baseName);
+    }
 
     function traverseDir(dir: string) {
         fs.readdirSync(dir).forEach(file => {
             let fullPath = path.join(dir, file);
+            
             if (fs.lstatSync(fullPath).isDirectory()) {
-                traverseDir(fullPath);
+                // Skip ignored directories
+                if (!shouldIgnore(fullPath)) {
+                    traverseDir(fullPath);
+                }
             } else {
+                // Skip non-TypeScript/JavaScript files
+                if (!/\.(ts|tsx|js|jsx)$/.test(fullPath)) {
+                    return;
+                }
+
+                // If Git tracking is enabled and file is not tracked, skip it
+                if (gitAvailable && !gitTrackedFiles.has(path.resolve(fullPath))) {
```

Comment:
This check is problematic in the case of new files not yet added to git.  If running one of the commands that invokes this (init, audit) on a directory with uncommited files those new files would be skipped which could give an unexpected false result.

Reply:
There's a different way to do it so that uncommited files can still be picked up.  

I found the `git check-ignore` command which seems more suitable for this situation.  That command tells you whether a file is ignored by git or not which also covers new files that aren't tracked yet (assuming they're not ignored based on gitignore rules).